MatrixKit/Utils/EventFormatter/MarkdownParser.swift

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2015 OpenMarket Ltd
+Copyright 2017 Vector Creations Ltd
+Copyright 2018 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+import Down
+import libcmark
+
+@objcMembers
+public class MarkdownParser: NSObject {
+    
+    private let markdown: String
+    fileprivate var options: DownOptions
+    
+    public init(withMarkdown markdown: String) {
+        self.markdown = markdown
+        self.options = .default
+        super.init()
+    }
+    
+    public func toHTML() -> String? {
+        do {
+            return try Down(markdownString: markdown).toHTML(options)
+        } catch {
+            return nil
```

Comment:
It could be interesting to log error here but not leaking message, we can use "[MarkdownToHTMLRenderer] fail to render to HTML" for example. Otherwise prefer use `try?` in this case in order to return in one line: `return try? Down(markdownString: markdown).toHTML(options)`.

Reply:
Updated